### PR TITLE
Use installed flow version to get compatibiltiies from matrix

### PR DESCRIPTION
**Pull request by pankajkoti**
## Description

> Currently, we get the constraint of the SQL CLI python package that is used for installing flow and use the same constraint as a key to match in the compatibility matrix. However the compatibility matrix is meant to contain specific Pypi versions as key and not the constraints itself. We therefore need to get the installed flow version and use that flow version to find the compatibilities. For this reason, we call the `flow version` command which gives us the installed version of flow and we use the same to match it in the matrix.

## 🎟 Issue(s)

closes: https://github.com/astronomer/astro-sdk/issues/1642

## 🧪 Functional Testing

> List the functional testing steps to confirm this feature or fix.

## 📸 Screenshots

> Add screenshots to illustrate the validity of these changes.

## 📋 Checklist

- [ ] Rebased from the main (or release if patching) branch (before testing)
- [ ] Ran `make test` before taking out of draft
- [ ] Ran `make lint` before taking out of draft
- [ ] Added/updated applicable tests
- [ ] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: cmd/sql/flow.go [hunk 1]
```diff
@@ -24,19 +24,21 @@ import (
...
+	flowCmdName            = "flow"
+	aboutCmdName           = "about"
+	configCmdName          = "config"
+	generateCmdName        = "generate"
+	initCmdName            = "init"
+	runCmdName             = "run"
+	validateCmdName        = "validate"
+	versionCmdName         = "version"
+	deployCmdName          = "deploy"
+	versionCmdOutputPrefix = "Astro SQL CLI"
```
Comment (feluelle) [inline cmd/sql/flow.go:41]: Can we move that to its own `const` section?

**Review by feluelle — APPROVED** [review]
LGTM

File: cmd/sql/flow.go (hunk 1 above)
Comment (pankajkoti) [inline cmd/sql/flow.go:41]:
```suggestion
```

Comment (pankajkoti) [reply cmd/sql/flow.go]: This is no longer needed. We're now reading the json response.

**Review by kushalmalani — APPROVED** [review]

**Pull request merged**